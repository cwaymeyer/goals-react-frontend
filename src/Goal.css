.Goal {
  text-align: center;
  text-decoration: none;
}
