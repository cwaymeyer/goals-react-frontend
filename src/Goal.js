import { Link } from "react-router-dom";
import { Grid, Paper, Button } from "@mui/material";

function Goal({ data }) {
  return (
    <Link to={`/${data.id}`}>
      <Paper elevation={6} sx={{ height: 100, p: 2 }}>
        <Grid container>
          <Grid item xs={12} top={0}>
            <h3>{data.name}</h3>
          </Grid>
        </Grid>
      </Paper>
    </Link>
  );
}

export default Goal;
