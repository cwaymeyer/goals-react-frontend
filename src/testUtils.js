import UserContext from "./context/userContext";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const progress = [
  {
    id: 10,
    goal_id: 1,
    weight: 50,
    reps: 5,
    orm: "56.3",
    date: "1645477469350",
  },
  {
    id: 11,
    goal_id: 1,
    weight: 50,
    reps: 6,
    orm: "58.1",
    date: "1645477669350",
  },
];

const starting_progress = {
  orm: "56.3",
  date: "1645477269350",
};

const latest_progress = {
  orm: "58.1",
  date: "1645477669350",
};

const goal = {
  id: "1",
  name: "Test Goal",
  username: "demoUser",
  start_weight: null,
  target_weight: 100,
  timeline: 3,
  start_date: "1645477269350",
  end_date: "1653163269350",
  progress: progress,
  starting_progress: starting_progress,
  latest_progress: latest_progress,
};

const demoUser = {
  username: "demoUser",
  password: "password",
  goals: [goal],
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const renderWithRouter = (ui, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return render(ui, { wrapper: BrowserRouter });
};

export {
  UserProvider,
  renderWithRouter,
  goal,
  starting_progress,
  latest_progress,
};
