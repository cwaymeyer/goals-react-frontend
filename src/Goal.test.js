import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider, goal } from "./testUtils";
import Goal from "./Goal";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Goal data={goal} />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Goal data={goal} />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
