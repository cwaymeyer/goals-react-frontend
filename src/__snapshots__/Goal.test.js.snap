// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Goal"
  >
    <a
      href="/1"
    >
      <div
        class="paper"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1a1n6j7-MuiPaper-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ehlm6r-MuiGrid-root"
            >
              <h3>
                Test Goal
              </h3>
            </div>
          </div>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
